chartpress.py

```diff
@@ -727,7 +779,23 @@ def build_chart(
 
     # decide a version string
     if version is None:
-        version = _get_identifier_from_paths(*paths, long=long)
+        if use_chart_version:
+            # avoid adding .git... to chart version multiple times!
+            base_version = _trim_version_suffix(chart["version"])
+            # TODO: trim -set.by.chartpress?
+            # if base_version.endswith("-set.by.chartpress"):
+            #     base_version = base_version.rsplit("-", 1)[0]
```

Comment:
My understanding is that this is a failsafe to avoid situations when Chart.yaml has been updated with a chartpress version suffix already.

In the past, running chartpress multiple times wouldn't cause longer and longer version strings as  the resulting Chart.yaml specified version wouldn't be used as input. But, it is now. Makes sense.

---

@minrk I think the suggestion below would make me more comfortable with the changes overall as I think it would reduce risk of incorrect use of chartpress and reduce complexity of chartpress code etc.

1. I suggest we don't trim any `-set.by.chartpress` stuff, but that we opt to error if we observer that the version string is matching `resetVersion`. That would indicate that use_chart_version is configured but a Chart.yaml file don't have a version specified yet.

2. I suggest that whenever use_chart_version is specified, we assert that Chart.yaml includes a pre-release version, and not a stable version.

   Consider the `_trim_version_suffix` function. Chartpress can appends `0.dev` automatically, but we don't trim that.
   
   Consider version sorting. If you have a stable version in Chart.yaml and haven't "returned to dev", then we end up with the same issue as before where version ordering would fail.

   These issues would go away if we don't let chartpress also append `0.dev` sometimes, but instead just require that when `use_chart_version` is set we also expect a pre-release to be specified in Chart.yaml. I think this would reduce complexity without restricting users, it just enforces the intended use I think.

Reply:
I think I've implemented both of these with informative errors. I'm not sure that I agree that things are simplified by it, but there are fewer cases that are allowed to proceed without errors, at least.